ReactSkia/sdk/OnScreenKeyBoard.h

```diff
@@ -0,0 +1,158 @@
+    SkPoint        textHLXY{}; // Text X,Y to draw on Highlight Tile
+    SkPoint        textCapsHLXY{}; // Text X,Y for Upper Case Alphabets
+    SkRect         highlightTile; // Highlight Tile coverage
+}keyPosition_t;
+
+typedef struct KeyInfo {
+    const char *   keyName;
+    rnsKey         keyValue;
+    keyType        keyType;
+    unsigned int   kbPartitionId;
+}KeyInfo_t;
+
+typedef struct KeySiblingInfo {
+    SkPoint siblingRight;
+    SkPoint siblingLeft;
+    SkPoint siblingUp;
+    SkPoint siblingDown;
+}keySiblingInfo_t;
+
+typedef struct keyPlacementConfig {
+    SkPoint groupOffset;
+    SkPoint groupKeySpacing;
+    float   hlTileFontSizeMultiplier;
+    float   fontScaleFactor;
+    unsigned int maxTextLength;
+}keyPlacementConfig_t;
+
+typedef std::vector<std::vector<KeyInfo_t>> KBLayoutKeyInfoContainer;
+typedef std::vector<std::vector<keyPosition_t>> KBLayoutKeyPosContainer;
+typedef std::vector<std::vector<keySiblingInfo_t>> KBLayoutSibblingInfoContainer;
+
+
+struct OSKLayout {
+    KBLayoutKeyInfoContainer*  keyInfo;
+    KBLayoutKeyPosContainer*    keyPos;
+    KBLayoutSibblingInfoContainer*    siblingInfo;
+    keyPlacementConfig_t*          kbGroupConfig;
+    KBLayoutType      kbLayoutType;
+    unsigned int      textFontSize;
+    unsigned int      textHLFontSize;
+    unsigned int      horizontalStartOffset;
+    SkPoint           defaultFocussIndex;
+};
+
+class OnScreenKeyboard : public WindowDelegator{
+
+    public:
+        static OnScreenKeyboard& getInstance(); // Interface to get OSK singleton object
+        static OSKErrorCode launch(OSKConfig oskConfig=defaultOSKConfig);// Interface to launch OSK
+        static void exit(); //Interface to quit OSK
+        static void updatePlaceHolderString(std::string TIDisplayString);
+
+    private:
+        OnScreenKeyboard(){};
+        ~OnScreenKeyboard(){};
+
+        void launchOSKWindow(OSKConfig oskConfig);
+        void onHWkeyHandler(rnsKey key, rnsKeyAction eventKeyAction);
+        void createOSKLayout(OSKTypes KBtype );
+
+        void emitOSKKeyEvent(rnsKey keyValue);
+        void windowReadyToDrawCB();
+
+        void drawHighLightOnKey(SkPoint index);
+        void drawOSK();
+        void drawKBLayout(OSKTypes oskType);
+        void drawKBKeyFont(SkPoint index,SkColor color,bool onHLTile=false);
+
+// Members for OSK Layout & sytling
+        OSKConfig     oskConfig_;
+        OSKLayout     oskLayout_;
+        SkSize        screenSize_{0,0};
+        SkColor       bgColor_{SK_ColorWHITE};
+        SkColor       fontColor_{SK_ColorWHITE};
+
+// Members for OSK operations
+        int oskEventId_{-1};
+        bool          generateOSKLayout_{true};
+        SkPoint       currentFocussIndex_{};
+        SkPoint       lastFocussIndex_{};
+        std::string   displayString_{}; // Text to be displayed on screen
+        std::string   lastDisplayedString_{};
+        OSKState      oskState_{OSK_STATE_INACTIVE};
+};
+#endif //OSK_H
+
```

Comment:
Can we add namespace and move struct inside the parent class

Reply:
Can I Handle this as part of CRN-270[embeded textinput], which is awaiting for PR, Just to unblock it.